src/core/components/key-set.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
```

Comment:
I did it this way to work around the problem mentioned in the comment
```
  // Buffers with identical data are still different objects, so
  // they need to be cast to strings to prevent duplicates in Sets
```
in order to have set-like storage for multihash buffers but indexed by hash string. But since it's straightforward to convert the hash string back to the buffer object, maybe it would be better to just use a Set to store the strings and use `multihashes.fromB58String()` where the object is needed.